### PR TITLE
Electron cash update

### PR DESCRIPTION
**Pull request by nyanloutre**
###### Motivation for this change

Fix failed build

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by nyanloutre** [comment]
Can be backported to 19.03 to fix the build

File: pkgs/applications/misc/electron-cash/default.nix [hunk 1]
```diff
@@ -7,14 +7,14 @@ let
 in
 
 python3Packages.buildPythonApplication rec {
-  version = "3.3.2";
+  version = "3.3.6";
   name = "electron-cash-${version}";
```
Comment (dotlambda) [inline pkgs/applications/misc/electron-cash/default.nix]:
```suggestion
  name = "electron-cash";
```

File: pkgs/applications/misc/electron-cash/default.nix [hunk 2]
```diff
@@ -40,26 +41,22 @@ python3Packages.buildPythonApplication rec {
...
   postInstall = ''
-    # These files are installed under $out/homeless-shelter ...
-    mv $out/${python.sitePackages}/homeless-shelter/.local/share $out
-    rm -rf $out/${python.sitePackages}/homeless-shelter
-
     substituteInPlace $out/share/applications/electron-cash.desktop \
       --replace "Exec=electron-cash %u" "Exec=$out/bin/electron-cash %u"
 
+    substituteInPlace $out/share/applications/electron-cash.desktop \
+      --replace "Exec=electron-cash --testnet %u" "Exec=$out/bin/electron-cash --testnet %u"
```
Comment (dotlambda) [inline pkgs/applications/misc/electron-cash/default.nix]:
```suggestion
      --replace "Exec=electron-cash" "Exec=$out/bin/electron-cash"
```

File: pkgs/development/python-modules/qdarkstyle/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchPypi, buildPythonPackage }:
+
+buildPythonPackage rec {
+  pname = "QDarkStyle";
```
Comment (dotlambda) [inline pkgs/development/python-modules/qdarkstyle/default.nix]:
```suggestion
  pname = "qdarkstyle";
```
The line should instead be put inside `fetchPypi`.

File: pkgs/development/python-modules/qdarkstyle/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchPypi, buildPythonPackage }:
```
Comment (dotlambda) [inline pkgs/development/python-modules/qdarkstyle/default.nix]:
```suggestion
{ lib, fetchPypi, buildPythonPackage }:
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by nyanloutre** [comment]
@dotlambda I applied your suggestions, thanks

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.qdarkstyle python3.pkgs.qdarkstyle

**Comment by dotlambda** [comment]
@GrahamcOfBorg build electron-cash

**Comment by dotlambda** [comment]
cherry picked in 3eb63d12194 and 97042d883f3

**Pull request merged**